Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py

```diff
@@ -521,23 +592,30 @@ def main():  # pragma: no cover
         elif command == "fetch-events":
             page_size = int(params.get("page_size", FETCH_EVENTS_PAGE_SIZE))
             limit = int(params.get("fetchLimit", 300000))
-            for events, offset, total_events_count, hashed_events_from_current_run in fetch_events_command(  # noqa: B007
+            if limit > MAX_ALLOWED_FETCH_LIMIT:
+                demisto.info(f"Got {limit=} larger than {MAX_ALLOWED_FETCH_LIMIT=}, setting limit to {MAX_ALLOWED_FETCH_LIMIT}.")
+                limit = MAX_ALLOWED_FETCH_LIMIT
+            if limit < page_size:
+                demisto.info(f"Got {limit=} lower than {page_size=}, lowering page_size to {limit}.")
+                page_size = limit
+            for events, offset, total_events_count, hashed_events_from_current_run, auto_trigger_next_run in (  # noqa: B007
+            fetch_events_command(
                 client,
                 "5 minutes",
                 fetch_limit=limit,
                 config_ids=params.get("configIds", ""),
                 ctx=get_integration_context() or {},
                 page_size=page_size
-            ):
+            )):
                 if events:
                     demisto.info(f"Sending events to xsiam with latest event time is: {events[-1]['_time']}")
                     send_events_to_xsiam(events, VENDOR, PRODUCT, should_update_health_module=False)
                 set_integration_context({"offset": offset,
                                          "hashed_events_from_previous_run": list(hashed_events_from_current_run)})
             demisto.updateModuleHealth({'eventsPulled': (total_events_count or 0)})
             next_run = {}
-            if total_events_count >= limit:
-                demisto.info(f"got at least {limit} events this interval - will automatically trigger next run.")
+            if auto_trigger_next_run or total_events_count >= limit:
```

Comment:
shouldnt we check if its within the ratio? Meaning getting 44999 events should still trigger next run

Reply:
If the request returned a page lower by delta then we don't want to trigger it.
However, if there was a break due to being afraid of timeouts then auto_trigger_next_run should be True.

Anyway, we should be covered for this one.